src/main/io/osd.c

```diff
@@ -227,16 +227,64 @@ static uint8_t osdGetDirectionSymbolFromHeading(int heading)
     return SYM_ARROW_SOUTH + heading;
 }
 
+// this will convert from relative positioning (i.e. measured from center pos)
+// to absolute positioning (based on display height and width)
+static void osdConvertToAbsolutePosition(uint8_t item, int8_t *pos_x, int8_t *pos_y) {
+    // output display dimensions
+    uint8_t display_width  = osdDisplayPort->colCount - 1;
+    uint8_t display_height = osdDisplayPort->rowCount - 1;
+
+    // x/y position (note: might need origin offset, see below)
+    int8_t elemPosX = osdConfig()->item_pos[item].x;
+    int8_t elemPosY = osdConfig()->item_pos[item].y;
+
+    // fetch origin positio
+    uint8_t origin = osdConfig()->item_pos[item].origin;
+
+    // start with center
+    elemPosX += display_width / 2;
```

Comment:
You may start with zero, add `width/2` if not west and another  `width/2` if east